### PR TITLE
CI: Add job to test on Linux

### LINKED ISSUE #795: Tests not passing in Linux
I just ran our tests in Linux (see #791) and it seems that a part of our code does not pass the tests when they are run on Linux.

Here are the two tests failing:
- [x] tests/legacy/system/test_edb.py::TestClass::test_siwave_dc_simulation_setup
- [x] tests/legacy/system/test_edb.py::TestClass::test_move_and_edit_polygons

@svandenb-dev @ring630 Could one of you handle this ?

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
As title says.

Close #795

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
There seem to be a strange behavior when working with `dc_advanced_settings` in a `setup` as `test_siwave_dc_simulation_setup` fails randomly **in Linux**. Since there are already other key words that are "black listed" (seems like `compute_inductance` and `plot_jv` have the same problem) I added `dc_advanced_settings` to the skip list. 

@ring630 @svandenb-dev It might be worth to report a bug in TFS if this is related to EDB and not `pyedb`. Can you handle that ?

> **Note**: this is not related to the testing environment since the following snippet fails/passes randomly:

```python
from pyedb import Edb

def main():
    app = Edb("dummy_name")
    setup = app.create_siwave_dc_setup("DC1")
    setup.dc_settings.restore_default()
    setup.dc_advanced_settings.restore_default()
    settings = app.setups["DC1"].get_configurations()
    for k, v in setup.dc_settings.defaults.items():
        if k in ["compute_inductance", "plot_jv"]:
            continue
        print(k)
        assert settings["dc_settings"][k] == v
    print(settings["dc_settings"]["use_dc_custom_settings"] == False)
    app.close_edb()
    del app

main()
```

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -133,6 +133,75 @@ jobs:
...
+        run: |
+          . .venv/bin/activate
+          python -m pip install .[tests]
+
+      - name: "Executing legacy tests"
+        uses: nick-fields/retry@v3
+        with:
+          max_attempts: 3
+          retry_on: error
+          timeout_minutes: 15
```
Comment (SMoraisAnsys) [inline .github/workflows/ci_cd.yml:179]:
```suggestion
          timeout_minutes: 20
```

**Comment by SMoraisAnsys** [comment]
@ring630 @svandenb-dev  Previously the Linux tests were passing but they are no longer passing since a merged `main` into this branch. Here is the diff https://github.com/ansys/pyedb/commit/ff3fdd7229564191cdf114447dbac850b729f180 that was added, it might related to the failure in pyaedt so I took the liberty to ping you guys.

**Comment by ring630** [comment]
> There seem to be a strange behavior when working with `dc_advanced_settings` in a `setup` as `test_siwave_dc_simulation_setup` fails randomly **in Linux**. Since there are already other key words that are "black listed" (seems like `compute_inductance` and `plot_jv` have the same problem) I'm adding `dc_advanced_settings` to the skip list.
> 
> @ring630 @svandenb-dev It might be worth to report a bug in TFS if this is related to EDB and not `pyedb`. Can you handle that ?
> 
> > **Note**: this is not related to the testing environment since the following snippet fails/passes randomly:
> 
> ```python
> from pyedb import Edb
> 
> def main():
>     app = Edb("dummy_name")
>     setup = app.create_siwave_dc_setup("DC1")
>     setup.dc_settings.restore_default()
>     setup.dc_advanced_settings.restore_default()
>     settings = app.setups["DC1"].get_configurations()
>     for k, v in setup.dc_settings.defaults.items():
>         if k in ["compute_inductance", "plot_jv"]:
>             continue
>         print(k)
>         assert settings["dc_settings"][k] == v
>     print(settings["dc_settings"]["use_dc_custom_settings"] == False)
>     app.close_edb()
>     del app
> 
> main()
> ```

The Dev added SimSetupInfo for SIwave in 25.1. I hope Dev fixed this issue in SimSetupInfo. However to test it we need to refactor SimulationSetup class.

**Review by ring630 — APPROVED** [review]
LGTM

**Pull request merged**